filters/auth/oidc.go

```diff
@@ -327,6 +333,12 @@ func (f *tokenOidcFilter) doOauthRedirect(ctx filters.FilterContext) {
 		return
 	}
 
+	for _, param := range f.queryParams {
+		if queryValue := ctx.Request().URL.Query().Get(param); len(queryValue) > 0 {
+			f.authCodeOptions = append(f.authCodeOptions, oauth2.SetAuthURLParam(param, queryValue))
```

Comment:
Here filter setting are modified so it will grow on each call, it should be a copy, something like 
```diff
-       oauth2URL := f.config.AuthCodeURL(fmt.Sprintf("%x", stateEnc), f.authCodeOptions...)
+       opts := f.authCodeOptions
+       if f.queryParams != nil {
+               opts = make([]oauth2.AuthCodeOption, 0, len(f.authCodeOptions)+len(f.queryParams))
+               copy(opts, f.authCodeOptions)
+               for _, p := range f.queryParams {
+                       if v := ctx.Request().URL.Query().Get(p); v != "" {
+                               opts = append(opts, oauth2.SetAuthURLParam(p, v))
+                       }
+               }
+       }
+
+       oauth2URL := f.config.AuthCodeURL(fmt.Sprintf("%x", stateEnc), opts...)
```

Reply:
Great! That is much better. I pushed the changes. Thank you.